internal/intake/worker.go

```
@@ -181,7 +190,15 @@ func (w *worker) sendDelta(ctx context.Context) {
 	}
 	defer w.queue.Done(delta)
 
-	if w.stream == nil {
+	if w.stream == nil || time.Since(w.streamCreatedAt) > w.maxStreamAge {
```

I believe another you can do this is set a timeout on the context that you pass into the stream. So below we have
```golang
streamCtx := metadata.NewOutgoingContext(
	context.Background(), metadata.Pairs(headerAuthorize, fmt.Sprintf("bearer %s", w.apiKey)),
)
stream, err := w.client.Delta(streamCtx)
if err != nil {
	w.logger.Error(err, "failed to create intake stream, retrying...")
	return false, err
}
```

you could replace `context.Background` with `context.WithTimeout(context.Background, maxStreamAge)` which would essentially set the lifetime of the stream.

When the stream times out, `stream.Send` should return an error indicating that the stream was canceled at which point you can create a new stream

Here's an example https://github.com/grpc/grpc-go/blob/64a6b623bab1371a08e934609abca29ff1dbd668/examples/features/cancellation/client/main.go

---

switched to use that now. there was a complication with the cancel() method that get's reaturned along with it.